internals/overlord/servstate/manager.go

```diff
@@ -334,6 +335,20 @@ func (m *ServiceManager) CheckFailed(name string) {
 	}
 }
 
+// WriteMetrics collects and writes metrics for all services to the provided writer.
+func (m *ServiceManager) WriteMetrics(writer metrics.Writer) error {
+	m.servicesLock.Lock()
+	defer m.servicesLock.Unlock()
+
+	for _, service := range m.services {
```

Comment:
It's probably best to make the order stable, so let's grab the service names into a slice and `sort.Strings` them before looping. Otherwise the order of `/v1/metrics` will change each time you refresh it, which is not ideal.

Similar for the checks `WriteMetrics`.